java/src/main/java/org/whispersystems/signalservice/loki/api/LokiGroupChatAPI.kt

```diff
@@ -65,22 +65,22 @@ public class LokiGroupChatAPI(private val userHexEncodedPublicKey: String, priva
                     200 -> {
                         try {
                             val bodyAsString = response.body()!!.string()
-                            @Suppress("NAME_SHADOWING") val body = JsonUtil.fromJson(bodyAsString, Map::class.java)
-                            val base64EncodedChallenge = body["cipherText64"] as String
+                            val root = JsonUtil.fromJson(bodyAsString)
+                            val base64EncodedChallenge = root.get("cipherText64").asText()
```

Comment:
If you do this you also need to check that the value returned by `asText()` is not empty. Currently the way this code works is that it relies on an error to be thrown if the JSON being parsed is invalid. This change breaks that pattern.

Reply:
Right but it should be fine because `decrypt` would fail anyway since the data is empty. Also from my testing, if the field `cipherText64` didn't exist then it would throw an error, it would only return an empty string if `cipherText64` was set to null.